#!/usr/bin/env python
PACKAGE = "victoria_perception"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# Parameters for the first HSV filter.
gen.add("alow_hue_",             int_t,    0, "Low Hue",           4,      0,    179)
gen.add("ahigh_hue_",            int_t,    0, "High Hue",          34,     0,    179)
gen.add("alow_saturation_",      int_t,    0, "Low Saturation",    103,    0,    255)
gen.add("ahigh_saturation_",     int_t,    0, "High Saturation",   222,    0,    255)
gen.add("alow_value_",           int_t,    0, "Low Value",         0,      0,    255)
gen.add("ahigh_value_",          int_t,    0, "High Value",        240,    0,    255)

# Parameters for the second HSV filter.
gen.add("blow_hue_",             int_t,    0, "Low Hue",           4,      0,    179)
gen.add("bhigh_hue_",            int_t,    0, "High Hue",          34,     0,    179)
gen.add("blow_saturation_",      int_t,    0, "Low Saturation",    103,    0,    255)
gen.add("bhigh_saturation_",     int_t,    0, "High Saturation",   222,    0,    255)
gen.add("blow_value_",           int_t,    0, "Low Value",         0,      0,    255)
gen.add("bhigh_value_",          int_t,    0, "High Value",        240,    0,    255)

gen.add("min_cone_area_",       int_t,    0, "Min Cone Area",     500,    0,    76800)
gen.add("max_cone_area_",       int_t,    0, "Max Cone Area",     45000,  0,    76800)
gen.add("max_aspect_ratio_",    double_t, 0, "Max Aspect Ration", 0.85,   0.0,  1.0)
gen.add("poly_epsilon_",        double_t, 0, "Poly epsilon",      6.7,    0.0,  20.0)

exit(gen.generate(PACKAGE, "cone_detector_node", "ConeDetector"))